### PR TITLE
MVP for today

### PR DESCRIPTION
**Pull request by Bilguun1015**

### CODE REVIEW

File: ls8/cpu.py [hunk 1]
```diff
@@ -62,4 +88,21 @@ def trace(self):
...
+                value = self.ram_read(self.PC + 2)
+                self.ram_write(address, value)
+                self.PC += 3
+            elif IR == self.PRN:
+                num = self.ram_read(self.PC - 3)
+                print(num)
+                self.PC +=2
+            else:
+                print(f"Error: Unknown command: {IR}")
+                sys.exit(1)
```
Comment (brianilles) [inline ls8/cpu.py:163]: How could we improve this logic so it is more extensible?

**Review by brianilles — APPROVED** [review]
Great start.

File: ls8/cpu.py [hunk 2]
```diff
@@ -88,21 +98,36 @@ def trace(self):
...
+                    value = self.ram_read(operand_b)
+                    self.reg[address] = value
+                    self.PC += 3
+                else:
+                    print('Not Enough arguments')
             elif IR == self.PRN:
-                num = self.ram_read(self.PC - 3)
-                print(num)
-                self.PC +=2
+                if (IR >> 6) == 1:
```
Comment (brianilles) [inline ls8/cpu.py]: Cool use of bitwise operator `>>`

File: ls8/cpu.py [hunk 3]
```diff
@@ -8,61 +8,71 @@ class CPU:
...
+        self.memory[address] = value
 
     def alu(self, op, reg_a, reg_b):
         """ALU operations."""
 
         if op == "ADD":
             self.reg[reg_a] += self.reg[reg_b]
         #elif op == "SUB": etc
+        elif op == 'MUL':
+            self.reg[reg_a] *= self.reg[reg_b]
```
Comment (brianilles) [inline ls8/cpu.py:71]: Excellent and readable

File: ls8/cpu.py [hunk 4]
```diff
@@ -8,61 +8,71 @@ class CPU:
...
+                    num = comment_split[0].strip()
+
+                    if num == '':
+                        continue
+                    
+                    value = int(num, 2)
+
+                    self.memory[address] = value
+                    address += 1
+        except FileNotFoundError:
```
Comment (brianilles) [inline ls8/cpu.py:54]: Great use of `try-except`

**Review by brianilles — APPROVED** [review]
Superb work on day 2

File: ls8/cpu.py [hunk 5]
```diff
@@ -83,51 +75,66 @@ def trace(self):
...
+        value = self.ram_read(operand_b)
+        self.reg[address] = value
+        self.PC += 3
+
+    def handle_PRN(self):
+        address = self.ram_read(self.PC + 1)
+        print(self.reg[address])
+        self.PC +=2
+
+    def handle_MUL(self):
```
Comment (brilles) [inline ls8/cpu.py]: Where can we put this to compose the code better? Hint: there's a special place for arithmetic operations...

File: ls8/cpu.py [hunk 6]
```diff
@@ -83,51 +75,66 @@ def trace(self):
...
-                    print(self.reg[address])
-                    self.PC +=2
-                else:
-                    print('More than 1 argument given')
-            else:
-                print(f"Error: Unknown command: {IR}")
-                sys.exit(1)
\ No newline at end of file
+            try:
+                return_command = self.branchtable[IR]()
```
Comment (brianilles) [inline ls8/cpu.py:158]: Great use of the branch table.

**Review by brianilles — APPROVED** [review]

**Review by brianilles — APPROVED** [review]
💯

**Pull request merged**